src/js/components/TaskDetail.js

```diff
@@ -151,56 +151,119 @@ class TaskSidePanelContents extends SidePanelContents {
+        {this.tabs_getUnroutedTabs()}
+      </ul>
+    );
 
     return (
-      <div>
-        <div className="side-panel-content-header container-fluid container-pod flush-top flush-bottom">
-          <h1 className="side-panel-content-header-label flush">
-            {task.name}
-          </h1>
-          <div className="media-object-spacing-wrapper media-object-spacing-narrow media-object-offset">
-            <div className="media-object media-object-align-middle">
-              <div className="media-object-item">
-                {statusIcon}
-              </div>
+      <PageHeader
+        icon={taskIcon}
+        subTitle={TaskStates[task.state].displayName}
+        navigationTabs={tabs}
+        mediaWrapperClassName="media-object-spacing-wrapper"
+        title={task.getId()} />
+    );
+  }
 
-              <div className="media-object-item">
-                <span className={statusClassName}>
-                  {TaskStates[task.state].displayName}
-                </span>
-              </div>
+  getTaskStatus(task) {
+    if (task == null || task.status == null) {
+      return 'Unknown';
+    }
+    return task.status;
+  }
 
-              <div className="media-object-item">
-                <span className="side-panel-subheader side-panel-subheader-emphasize">
-                  {node.hostname}
-                </span>
-              </div>
-            </div>
-          </div>
-        </div>
-        <div className="container container-pod container-pod-short container-fluid flush">
-          {this.getResources(task)}
-        </div>
-      </div>
+  getTaskEndpoints(task) {
+    let service = MarathonStore.getServiceFromTaskID(task.id);
+
+    if ((task.ports == null || task.ports.length === 0) &&
+        (task.ipAddresses == null || task.ipAddresses.length === 0)) {
+      return (<dd>None</dd>);
+    }
+
+    if (service != null &&
+        service.ipAddress != null &&
+        service.ipAddress.discovery != null &&
+        service.ipAddress.discovery.ports != null &&
+        task.ipAddresses != null &&
+        task.ipAddresses.length > 0) {
+
+      let ports = service.ipAddress.discovery.ports;
+      let endpoints = task.ipAddresses.reduce(function (memo, address) {
+        ports.forEach(port => {
+          memo.push(`${address.ipAddress}:${port.number}`);
+        });
+        return memo;
+      }, []);
+
+      if (endpoints.length) {
+        return endpoints.map(function (endpoint) {
+          return (
+            <a href={`//${endpoint}`} target="_blank">{endpoint}</a>
+          );
+        });
+      }
+
+      return 'n/a';
+    }
+
+    return task.ports.map(function (port) {
+      let endpoint = `${task.host}:${port}`;
+      return (
+        <a href={`//${endpoint}`} target="_blank">{endpoint}</a>
+      );
+    });
+  }
+
+  getMarathonTaskDetailsDescriptionList() {
+    let task = MarathonStore.getTaskFromTaskID(this.props.params.taskID);
```

Comment:
That task comes from the Marathon store not from the Mesos store.
I've renamed the var to clarify this.